FieldOpt/Model/model.cpp

```diff
@@ -105,6 +105,36 @@ void Model::verifyWells()
     }
 }
 
+Model::Economy Model::wellCost(Settings::Optimizer::Objective objective) {
+  Economy well_economy;
+  if (objective.use_well_cost) {
+    well_economy.n_wells = wells_->size();
+    well_economy.cost = objective.wellCost;
+    well_economy.costXY = objective.wellCostXY;
+    well_economy.costZ = objective.wellCostZ;
+    well_economy.separate = objective.separatehorizontalandvertical;
+    well_economy.use_well_cost = objective.use_well_cost;
+    well_economy.wells_ = *wells_;
+    for (Wells::Well *well : *wells_) {
+      auto variable = well->trajectory()->GetWellSpline()->GetSplinePoints();
```

Comment:
Although I see your point about well_xy, the use case you suggest here is not what it was designed for. Rather, it is specified in case you have different costs associated with drilling horizontal and vertical.